### PR TITLE
Support live photo

### PR DESCRIPTION
**Pull request by tmp-hallenser**
This PR adresses #378.

**DB changes:**
* new col 'LivePhotoURL' storing the URL to the video
* new col 'LivePhotoContentID' storing the contentID to match photo and video for Apple Live Photos (contentID is the unique identifier)

**Logic:**
1. Apple Live Photo
- Sequential upload of video and photo
- Matching based on LivePhotoContentID (exiftool mandatory to extract this)
- Eventually only a single db entry

2. Google Motion Photo
- Upload of the JPG (which contains the video)
- extraction of the video from the JPG file & convert it into a mov container (no recoding, just switch of container)
- Eventually only a single db entry

### CODE REVIEW

**Comment by d7415** [comment]
>     * extraction of the video from the JPG file & convert it into a mov container (no recoding, just switch of container)

Why mov? To keep it similar to the iOS version?

**Comment by tmp-hallenser** [comment]
> > ```
> > * extraction of the video from the JPG file & convert it into a mov container (no recoding, just switch of container)
> > ```
> 
> 
> Why mov? To keep it similar to the iOS version?

The npm package for the frontend (livephotokit, https://www.npmjs.com/package/livephotoskit) only supports mov. I did quite some research, but it seems to be the only package which is still maintained and working on current devices. Of course, we could build our own version supporting, but the benefit would be imho minimal.

Since we're keeping the same video and audio codec, it's just a different container and the conversion should be very fast (+ typically max 3 seconds of video)

**Comment by ildyria** [comment]
I will have a look at this PR once #401 is merged as the diff with master is quite big and redundant.

File: app/ModelFunctions/PhotoFunctions.php [hunk 1]
```diff
@@ -638,3 +755,34 @@ public function getValidExtensions(): array
 		return $this->validExtensions;
 	}
 }
+
+// Class for FFMpeg to convert files to mov format
+class MOVFormat extends FFMpeg\Format\Video\DefaultVideo
```
Comment (ildyria) [inline app/ModelFunctions/PhotoFunctions.php]: Why is this class not in another file ? :confused:

**Pull request merged**